### PR TITLE
Feature/baomt/t05 search job by title categories location

### PR DESCRIPTION
**Pull request by tminhbao**

### CODE REVIEW

File: src/components/SearchBar/SearchBar.tsx [hunk 1]
```diff
@@ -16,14 +21,29 @@ const SearchBar = (props: Props) => {
...
+      Swal.fire({
+        icon: "error",
+        title: "Oops...",
+        text: "You must type something before search",
+      });
+      return;
+    }
     const tempValue = searchValue.trim().toLowerCase();
-    dispatch(searchJobApi(tempValue));
+    if (searchOption === "1") {
```
Comment (TruongNP) [inline src/components/SearchBar/SearchBar.tsx:40]: Please use switch case instead of use "if/else"

File: src/components/SearchBar/SearchBar.tsx [hunk 2]
```diff
@@ -16,14 +21,29 @@ const SearchBar = (props: Props) => {
...
 
   const [searchValue, setSearchValue] = useState<string>("");
+  const [searchOption, setSearchOption] = useState<string>("1");
 
   useEffect(() => {
     dispatch(getAllJobsApi());
   }, [dispatch]);
 
   const handleSearchJob = () => {
+    if (!searchValue) {
```
Comment (TruongNP) [inline src/components/SearchBar/SearchBar.tsx:31]: Please show the text when search result is empty.
Not use  sweetalert2

File: src/redux/reducer/jobReducer.tsx [hunk 3]
```diff
@@ -59,13 +60,48 @@ export const paginationApi = (currentPage: number) => {
   };
 };
 
-export const searchJobApi = (searchValue: string) => {
+export const searchJobByTitleApi = (searchValue: string) => {
```
Comment (TruongNP) [inline src/redux/reducer/jobReducer.tsx:63]: Please use one search handler function instead of three

File: src/redux/reducer/jobReducer.tsx [hunk 4]
```diff
@@ -59,13 +60,48 @@ export const paginationApi = (currentPage: number) => {
...
   return async (dispatch: DispatchType) => {
     dispatch(changeIsLoading(true));
-    const result = await http.get(`/jobs/?search=${searchValue}`);
+    const result = await http.get(`/jobs/?jobTitle=${searchValue}`);
     if (result.data.length === 0) {
-      alert("Sorry, we do not find any jobs in the database");
-      window.location.href = "http://localhost:3000";
+      Swal.fire({
```
Comment (TruongNP) [inline src/redux/reducer/jobReducer.tsx:68]: Please show the text when search result is empty.

File: src/components/SearchBar/SearchBar.tsx (hunk 1 above)
Comment (tminhbao) [reply src/components/SearchBar/SearchBar.tsx:40]: Fixed

File: src/components/SearchBar/SearchBar.tsx (hunk 2 above)
Comment (tminhbao) [reply src/components/SearchBar/SearchBar.tsx:31]: Fixed

File: src/redux/reducer/jobReducer.tsx (hunk 3 above)
Comment (tminhbao) [reply src/redux/reducer/jobReducer.tsx:63]: Fixed

File: src/redux/reducer/jobReducer.tsx (hunk 4 above)
Comment (tminhbao) [reply src/redux/reducer/jobReducer.tsx:68]: Fixed

File: src/components/Jobs/JobList/JobList.tsx [hunk 5]
```diff
@@ -40,6 +40,19 @@ const JobList = (props: Props) => {
     );
   }
 
+  if (jobSearchList.length === 0 && jobPaginationList.length === 0) {
+    return (
+      <div className="container my-5">
```
Comment (TruongNP) [inline src/components/Jobs/JobList/JobList.tsx:45]: I think you should create a similar component with <Loading />
Ex: <Empty />

File: src/components/Jobs/JobList/JobList.tsx [hunk 6]
```diff
@@ -40,6 +40,19 @@ const JobList = (props: Props) => {
...
+          <p>Ops..! The job was not found !!</p>
+          <p>
+            Back to home: <a href="/">Home</a>
+          </p>
+        </div>
+      </div>
+    );
+  }
+
   if (jobSearchList.length > 0) {
```
Comment (TruongNP) [inline src/components/Jobs/JobList/JobList.tsx:56]: Do we need conditions here?
Same code below

File: src/components/Jobs/JobList/JobList.tsx (hunk 5 above)
Comment (tminhbao) [reply src/components/Jobs/JobList/JobList.tsx:45]: Fixed

File: src/components/Jobs/JobList/JobList.tsx (hunk 6 above)
Comment (tminhbao) [reply src/components/Jobs/JobList/JobList.tsx]: Fixed

**Review by TruongNP — APPROVED** [review]

**Pull request merged**